packages/server/src/integrations/mongodb.ts

```diff
@@ -141,18 +183,27 @@ module MongoDBModule {
       }
     }
 
-    async update(query: { json: object; extra: { [key: string]: string } }) {
+    async update(query: { json: UpdateDoc; extra: { [key: string]: string } }) {
```

Comment:
Instead of defining our own type here for `UpdateDoc` we can use the official mongo types. To do this there would only be a few changes needed

1. Add the dependency
```
yarn add --dev @types/mongodb@3.6.3
```

2. Update the mongo import, instead of `require` use `import`. Move this to the top level of the file outside of `MongoDBModule`
```
import { MongoClient, ObjectId } from "mongodb"
```

I tried this locally and there are a few req squiggly lines around our usage of the mongo client, similar to using the `UpdateDoc` fix, maybe there could be more

Reply:
I brought in the official types. 

Doing this made me realised I'm probably doing this wrong, but it might be okay for now? 

To do filter/updates with this implementation you need to do:
```
{
  "filter": { "name": "A" },
  "update": { "$set": { "name": "B" } }
}  
```
but if you were doing this as a mongodb query it should really be:
**Update One Selected**
```
{ "name": "A" },
{ "$set": { "name": "B" } }
```

Reply:
I see what you mean, having the two different dialects isn't ideal here, this will add translation from the bb query domain to the mongo one that'll need extra docs, etc. Probably the right way here is to have fields rendered based on the query type selected, that sounds like a v2 thing though. 

The simplest thing to do for now might be to allow them to paste an identical mongo query and try to determine which part is the operator object in the update queries `findOneAndUpdate`, `updateOne` and `updateMany`. Could be a simple check for `$` in the key, and fallback to the second object which seems to be the case in all the example docs.

Reply:
I've updated it to parse the params if needed. So both of the options above will now work.

Reply:
sounds great, good job mel 👍